hack/generate-authors.sh

```diff
@@ -3,7 +3,7 @@
 set -e
 
 SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
-ROOTDIR="$(git -C "$SCRIPTDIR" rev-parse --show-toplevel)"
+ROOTDIR="$(realpath "${SCRIPTDIR}/..")"
```

Comment:
Do we need to change this back to the variant we had before https://github.com/moby/moby/commit/b94d1604a9fc2a1656182eb206627697c12a5ca9 ?

```bash
hack/generate-authors.sh
hack/generate-authors.sh: line 6: realpath: command not found
```

Reply:
You should have a `/bin/realpath` on a mac; but yeah, we can move back to the 'stack builtins' method... It's just odd that realpath isn't available on your system.

Reply:
It's not available by default on macOS e.g. see https://stackoverflow.com/questions/3572030/bash-script-absolute-path-with-os-x

Reply:
I wonder if you're on an older version of macOS? macOS 13 seems to include `/bin/realpath` (/bin is in the sealed system partition, so it can't be part of Xcode/the Developer Tools); my manpage does not detail the inclusion history in macOS, alas:

```
HISTORY
     The realpath utility first appeared in FreeBSD 4.3.
```

However, `realpath` does appear to finally be available on every modern Unix-like system I have access to; would you be opposed to taking this as is (and updating your OS if that's the issue)?

Reply:
Yes, I'm on macOS 12 ("Monterey"), but that's still a supported version, so I don't think we should break that.

Reply:
Why not just use the same `$(cd "..." && pwd)` as the line above it that's doing the same thing? (macOS `pwd` might even support `-P` if you need symlinks resolved)

Reply:
oh, or even just `dirname`, since it looks like we're just looking to resolve `..`?

Reply:
Yeah, technically just `dirname` is fine; being more robust here is just trying to keep things working as expected if someone copies and pastes these files without understanding the limitation.

I think I probably will just revert things to how they were a couple months ago, stacking `readlink` with `dirname`, and add a TODO to use `realpath` once macOS 12 is N-2 versions.